csharp/src/Apache.Arrow.Adbc/Tracing/TracingConnection.cs

```diff
@@ -0,0 +1,60 @@
+﻿/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+
+namespace Apache.Arrow.Adbc.Tracing
+{
+    public abstract class TracingConnection : AdbcConnection, IActivityTracer
+    {
+        private const string SourceNameDefault = "apache.arrow.adbc";
+        private readonly ActivityTrace _trace;
+        private bool _isDisposed;
+
+        protected TracingConnection(IReadOnlyDictionary<string, string> properties)
+        {
+            properties.TryGetValue(AdbcOptions.Telemetry.TraceParent, out string? traceParent);
+            _trace = new ActivityTrace(this.GetAssemblyName(), this.GetAssemblyVersion(), traceParent);
```

Comment:
Consider storing the assembly name and version as statics to avoid having to compute them every time a connection is created.

Reply:
I can refactor the exporter to save the objects in a queue instead of serialize/deserialize. WIP.

Reply:
`SerializableActivity` is now removed from here.